File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        //count frequencies 
        int[] char_set = new int[26];
        
        if(s.length() != t.length()) return false;
        
        //rat
        //car
        for(int i=0; i<s.length(); i++){
           char_set[s.charAt(i)-'a']++;
           char_set[t.charAt(i)-'a']--;
        }
        
        
        for(int num : char_set){
            if(Math.abs(num) > 0) return false;
        }
        
        return true;
        
       
        
        
        
        
    }
}

File: 0242-valid-anagram/0242-valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
     if(s.length !== t.length) return false;

     var obj = {};
     for(let i=0; i<s.length; i++){
         obj[s[i]] = ( obj[s[i]] || 0 ) + 1;
     }

     for(let j=0; j<t.length; j++){
        if(!obj[t[j]] || obj[t[j]] === 0) return false;

        if(obj[t[j]]){
            obj[t[j]] = obj[t[j]] - 1;
        }
     }

     return true;
};